src/core/server/rendering/rendering_context_provider.test.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { mockCoreContext } from '../core_context.mock';
+import { httpServiceMock } from '../http/http_service.mock';
+import { httpServerMock } from '../http/http_server.mocks';
+import { pluginServiceMock } from '../plugins/plugins_service.mock';
+import { uiSettingsServiceMock } from '../ui_settings/ui_settings_service.mock';
+import { RenderingContextProvider } from './rendering_context_provider';
+
+describe('RenderingContextProvider', () => {
+  let context: ReturnType<typeof mockCoreContext.create>;
+  let http: ReturnType<typeof httpServiceMock.createSetupContract>;
+  let plugins: ReturnType<typeof pluginServiceMock.createSetupContract>;
+  let request: ReturnType<typeof httpServerMock.createKibanaRequest>;
+  let response: ReturnType<typeof httpServerMock.createResponseFactory>;
+  let uiSettings: ReturnType<typeof uiSettingsServiceMock.createClient>;
+  let provider: RenderingContextProvider;
+
+  beforeEach(() => {
+    context = mockCoreContext.create();
+    http = httpServiceMock.createSetupContract();
+    plugins = pluginServiceMock.createSetupContract();
+    request = httpServerMock.createKibanaRequest();
+    response = httpServerMock.createResponseFactory();
+    uiSettings = uiSettingsServiceMock.createClient();
+    provider = new RenderingContextProvider({
+      env: context.env,
+      http,
+      plugins,
+      request,
+      response,
+      uiSettings,
+    });
+
+    uiSettings.getUserProvided.mockResolvedValue({
+      providedBy: {
+        isOverridden: true,
+        value: 'user',
+      },
+    });
+  });
+
+  describe('#render()', () => {
+    it('renders "core" page', async () => {
+      await provider.render();
+      expect(response.ok).toHaveBeenCalled();
+    });
+
+    it('renders with custom injectedVarsOverrides', async () => {
+      await provider.render({ fake: '__TEST_TOKEN__' });
+      expect(response.ok).toHaveBeenCalled();
+    });
+
+    it('renders with excluded user settings', async () => {
```

Comment:
what does this test check?

Reply:
That the page still renders if you leave out the call to get the user settings.